app/frontend/components/module_assistant/assistant_helper.js

```diff
@@ -0,0 +1,145 @@
+ * @param {boolean} [options.info=false] display info icon and generate event on click
+ * @param {boolean} [options.extraIndent=false] add extra indent to make sure that all item icons will be visible
+ * @param {string} [options.info="0.5em"] CSS size for the gap between the rows
+ * @returns {DocumentFragment} HTML fragment with appropriate <assistant-checkbox> elements for each item
+ */
+module.exports.listCheckboxSection = function (arr, selected, sectionTitle="", options={}) {
+  if (arr.length === 0) {
+    return '';
+  }
+
+  options = {
+    columns: 'auto-fill',
+    disableSelected: false,
+    info: false,
+    extraIndent: false,
+    rowGap: '0.5em',
+    ...options
+  };
+
+  var checkboxes = [];
+  for (const item of arr) {
+    if (typeof item === 'string') {
+      checkboxes.push(`<assistant-checkbox code="${item}" ${selected.has(item) ? 'checked' : ''}>${item}</assistant-checkbox>`);
+    } else {
+      const {code, text, description, count, disabled, icon, ...rest} = item;
+      const checkedProp = selected.has(code);
+      const disabledProp = disabled || options.disableSelected && checkedProp;
+
+      const style = text && text.length > 22 && (options.columns === 'auto-fill' || options.columns > 1) ? 'style="grid-column-end: span 2"' : '';
+
+      const iconSpan = icon ? `<span slot="label-icon">${icon}</span>` : '';
+
+      const extraAttr = Object.entries(rest).map(([attr, val]) => `${attr}="${val}"`);
+
+      /**@type {import('./assistant_checkbox')} */
+      const checkbox = `
+        <assistant-checkbox 
+          ${style}
+          code="${code}" 
+          ${checkedProp ? 'checked' : ''}
+          ${disabledProp ? 'disabled' : ''}
+          ${count ? `count="${count}"` : ''}
+          ${description ? `description="${description}"` : ''}
+          ${options.info ? `info="${i18n.t("module-assistant.click-to-show-detailed-module-info")}"` : ''}
+          ${extraAttr.join(' ')}>
+          ${iconSpan}
+          <span slot="label-text">${text ? text : code}</span>
+        </assistant-checkbox>`;
+
+      if (count !== 0) {
+        checkboxes.push(checkbox);
+      }
+    }
+  }
+
+  const paddingLeft = options.extraIndent ? '1em' : '0';
+
+  const template = html`
+    <h3 style="margin: 1em 0 0;">${sectionTitle}</h3>
+    <div style="display: grid; grid-template-columns: repeat(${options.columns}, minmax(15em, 1fr)); grid-row-gap: ${options.rowGap}; grid-column-gap: 1em; grid-auto-flow: dense; padding: 0.5em 0 0.5em ${paddingLeft};">
+      ${checkboxes}
+    </div>`;
+  return template.content;
+};
+
+
+module.exports.localize = function(container) {
+  container.querySelectorAll('[i18n]').forEach(element => {
+    element.innerHTML = i18n.t(element.getAttribute('i18n'));
+  });
+};
+
+module.exports.localizeContainer = function(container) {
+  var moduleTypeText = "";
+  const moduleType = assistantController.get('moduleType');
```

Comment:
I feel like this helper module should be purely functional with no additional dependency to the `assistantController`. I would put the `moduleType` as parameter instead and let the caller worry about getting the right value.